keras_cv/models/segmentation/deeplab.py

```diff
@@ -82,110 +95,88 @@ def __init__(
                     "Supported premade backbones are: "
                     f'{supported_premade_backbone}, received "{backbone}"'
                 )
-            self._backbone_passed = backbone
+
             if backbone == "resnet50_v2":
-                backbone = ResNetV2(
-                    stackwise_filters=BACKBONE_CONFIG["ResNet50V2"][
-                        "stackwise_filters"
-                    ],
-                    stackwise_blocks=BACKBONE_CONFIG["ResNet50V2"]["stackwise_blocks"],
-                    stackwise_strides=BACKBONE_CONFIG["ResNet50V2"][
-                        "stackwise_strides"
-                    ],
-                    stackwise_dilations=BACKBONE_CONFIG["ResNet50V2"][
-                        "stackwise_dilations"
-                    ],
-                    include_rescaling=include_rescaling,
-                    include_top=False,
-                    name="resnet50v2",
-                    weights=parse_weights(backbone_weights, False, "resnet50v2"),
-                    pooling=None,
-                    **kwargs,
+                if backbone_weights and input_shape[-1] == 1:
```

Comment:
what if input_shape = None?
when we use `inputs=parse_model_inputs`, just get the channels info as well.

Reply:
If the the input shape isn't known, the initialization fails, since the Upsampling layers needs the size during instantiation. Can we have a `build()` method besides `__init__` for this, while doing functional subclassing? If so, we could avoid this

Reply:
you can call `build` inside `__init__`

Reply:
Awesome! I'll update it to use the input size and see if the model can be built with unknown sizes upfront. The upsampling blocks are making it a bit more difficult

Reply:
> when we use inputs=parse_model_inputs, just get the channels info as well.

Got the channels, but they're never used in the arch explicitly. The linter is complaining about an unused variable